src/librustc_driver/driver.rs

```diff
@@ -667,7 +667,9 @@ pub fn phase_2_configure_and_expand<F>(sess: &Session,
                     new_path.push(path);
                 }
             }
-            env::set_var("PATH", &env::join_paths(new_path).unwrap());
+            if let Ok(p) = env::join_paths(new_path) {
```

Comment:
> It would give an invalid PATH.

What? Of course it would become a valid `PATH` once you filter out all the parts that can’t be put into `PATH`. Maybe it would not be complete, but certainly valid.

What your current patch does, on the other hand, is making whatever function this code had not come into effect whenever there’s any path from *any* source that cannot be put into `PATH`.